### DISCUSSION TITLE
Is there a way to punch holes in the bar background?

### DISCUSSION THREAD

**Discussion by izeau** [FelixKratz/SketchyBar#274] [Q&A] [answered]
Hello,

I would love to be able to have a black bar to kind of hide the notch but be able to punch holes in it that would leave the desktop background image visible in some spots. Is there a way to do that? I found an issue that was kind of related (#39) but I guess the method used to draw the background has changed since then.

Cheers!

**Comment by FelixKratz** [comment]
You can make the bar transparent and use brackets as demonstrated here: https://github.com/FelixKratz/SketchyBar/discussions/221 which gives the same effect as punching holes in the bar background.

**Reply by izeau** [reply]
I was actually thinking holes for the items / brackets, as in this (crude) mockup:

![mockup](https://user-images.githubusercontent.com/838867/197828681-a8536a1e-4db9-4287-aed5-208463f2d7c5.png)

**Comment by izeau** [comment, marked as answer]
I had to [mess with the code](https://github.com/izeau/SketchyBar/commit/3f6dc7542dcb22fb49a3020590492be64b57ed56) a little bit but I think I managed to find a way. The screenshot looks a bit stupid but with the screen bezel it's great in my opinion! I guess the clipping mask could be made a floating-point value — or even animatable — with little effort. I'll look into it, and submit a PR if that's OK with you!

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/838867/197910508-38722179-a622-46d5-bfcf-f636e3a0e129.png">

**Reply by FelixKratz** [reply]
Looks great! As long as this property does not affect performance when deactivated I am all for it. If you create a PR I will give a detailed review and merge the feature.
